sycl/source/detail/queue_impl.cpp

```diff
@@ -339,10 +341,51 @@ event queue_impl::submit_impl(const detail::type_erased_cgfo_ty &CGF,
 
   HandlerImpl->MEventMode = SubmitInfo.EventMode();
 
-  auto Event = finalizeHandler(Handler, SubmitInfo.PostProcessorFunc());
+  auto isHostTask = Type == CGType::CodeplayHostTask;
+
+  // TODO: we shouldn't need to treat ExecCommandBuffer differently,
+  // but currently UR (legacy L0 adapter) does not respect
+  // in-order property of command buffer and still requires sync points.
+  auto isGraphSubmission = Type == CGType::ExecCommandBuffer;
```

Comment:
I created an issue with what I think needs to be done here on the UR side if you want to link it, or add extra info/discussion https://github.com/intel/llvm/issues/18330